### PR TITLE
feat: governor

### PR DESCRIPTION
**Pull request by notJoon**
# Description

Implement basic Governor contract structure and key features.

## Key Features

- `Propose`: create a new proposal
- `Describe`: add a description to a proposal
- `ProposeAndDescribe`: combines Propose and Describe
- `Vote`: vote on a proposal
- `Cancel`: cancel a proposal
- `Execute`: execute an approved proposal

### CODE REVIEW

File: governor/types.gno [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	// How many no votes have been collected
+	Nay *u256.Uint
+
+	// The version of the config that this proposal was created with
+	ConfigVersion uint64
+}
+
+// CallFunction prepresents a function that can be called by the governor
+type CallFunction func([]byte) ([]byte, error)
```
Comment (r3v4s) [inline governor/types.gno]: Quick question.
`func([]byte)` doesn't cover if more than 2 parameters are being passed.

File: governor/gov.gno [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+var (
+	timeNow   = time.Now
+	gGovernor *Governor
+)
+
+// Init initializes the governor contract with the given staker and configuration.
+// It creates a new Governor instance and sets it as the global governor.
+// This function should be called only once at the beginning of the contract's lifecycle.
+func Init(staker IStakerDispatcher, config Config) {
```
Comment (r3v4s) [inline governor/gov.gno:26]: Does it have to call manually?
Can't we just init them in `init()`

File: governor/gov.gno [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+		voter,
+		votingStartTime-config.VotingWeightSmoothingDuration,
+		votingStartTime,
+	)
+	if err != nil {
+		panic("FAILED_TO_GET_WEIGHT")
+	}
+
+	if yes {
+		proposal.Yea = u256.Zero().Add(proposal.Yea, weight)
```
Comment (r3v4s) [inline governor/gov.gno]: Check overflow if necessary

**Review by r3v4s — CHANGES_REQUESTED** [review]
- Test cases run good on current gno's master branch.
- Please check few comments

File: governor/gov.gno (hunk 3 above)
Comment (notJoon) [reply governor/gov.gno]: https://github.com/gnoswap-labs/governance/pull/5/commits/f3e30d3ba7fc3b0ef917cbeab16fbc6463537cd9

**Review by dongwon8247 — APPROVED** [review]

File: governor/types.gno (hunk 1 above)
Comment (notJoon) [reply governor/types.gno]: chaged to receive string slice as a parameter

https://github.com/gnoswap-labs/governance/pull/5/commits/91d7024d24973e2448887ec7d9f10655821e5e12

File: governor/gov.gno (hunk 2 above)
Comment (notJoon) [reply governor/gov.gno:26]: Using `init()` is quite hard to initialize multiple times or apply different settings, so I decided to do it manually for flexibility.

**Review by r3v4s — APPROVED** [review]

**Pull request merged**